nh.c

```diff
@@ -461,3 +467,57 @@ void nh_failsafe_uninstall(nh_failsafe_t *fs) {
     nh_log("(NickelHook) ... failsafe: info: deleting %s", fs->tmp);
     unlink(fs->tmp);
 }
+
+// --- File/directory deletion
+
+// The following blacklist helps prevent any whoopsies when deleting files or directories during uninstall.
+
+// Let's not allow deleting any files in /bin, /sbin, /etc/init.d or u-boot stuff.
+static const char *delete_prefix_blacklist[] = {
+    "/bin",
+    "/sbin",
+    // note, '/etc/u-boot/*/u-boot.recovery' is a critical file for recovery
+    "/etc/u-boot",
+    "/etc/init.d",
+    "/usr/local/Kobo/pickel",
+};
+
+// nh_delete_path deletes the provided path if it exists. It performs some checks
```

Comment:
same here

Reply:
Done. Also took the liberty of wrapping the comments above the blacklist as well.